### PR TITLE
Added support to add mentions without token

### PR DESCRIPTION
**Pull request by sagar-waghmare-rsl**
- Added support to add mentions to the EditText without actually
  replacing the token.
- This is useful when we don't have a token and we just want to
  insert the mention. One such example would be like - adding a
  recently choosen mention.
- Added function "MentionsEditText:insertMentionWithoutToken"
  and test case as well.

### CODE REVIEW

File: spyglass/src/androidTest/java/com/linkedin/android/spyglass/ui/MentionsEditTextTest.java [hunk 1]
```diff
@@ -97,4 +97,21 @@ private void testMention(String hello, Mentionable mention) throws Exception {
...
         assertTrue(mEditText.getText().toString().startsWith("Hello "));
     }
+
+    @Test
+    public void testInsertMentionWithoutToken() throws Exception {
+        Mentionable mention = new TestMention("FirstName MiddleName LastName");
+        mEditText.insertMentionWithoutToken(mention);
+
+        // ensure mention adds correctly
+        assertTrue(mEditText.getText().toString().startsWith("FirstName MiddleName LastName"));
```
Comment (nhibner) [inline spyglass/src/androidTest/java/com/linkedin/android/spyglass/ui/MentionsEditTextTest.java]: Is there any particular reason why you're using "startsWith" and not "equals" here?

File: spyglass/src/main/java/com/linkedin/android/spyglass/ui/MentionsEditText.java [hunk 2]
```diff
@@ -673,6 +673,41 @@ public void insertMention(Mentionable mention) {
...
+        // Hide the suggestions and clear adapter
+        if (mSuggestionsVisibilityManager != null) {
+            mSuggestionsVisibilityManager.displaySuggestions(false);
+        }
+
+        // Reset input method since text has been changed (updates mention draw states)
+        InputMethodManager imm = (InputMethodManager) getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
+        if (imm != null) {
+            imm.restartInput(this);
+        }
```
Comment (nhibner) [inline spyglass/src/main/java/com/linkedin/android/spyglass/ui/MentionsEditText.java]: This code has a lot in common with the previous function. Can you please extract the common functionality into a private function they can both use, avoiding duplicate code?

File: spyglass/src/androidTest/java/com/linkedin/android/spyglass/ui/MentionsEditTextTest.java (hunk 1 above)
Comment (sagar-waghmare-rsl) [inline spyglass/src/androidTest/java/com/linkedin/android/spyglass/ui/MentionsEditTextTest.java]: Fixed this. We've used equals now.

File: spyglass/src/main/java/com/linkedin/android/spyglass/ui/MentionsEditText.java (hunk 2 above)
Comment (sagar-waghmare-rsl) [inline spyglass/src/main/java/com/linkedin/android/spyglass/ui/MentionsEditText.java]: Moved this common code to insertMentionInternal.

**Pull request merged**